docs/specs/src/hwp_4_formal.md

```diff
@@ -0,0 +1,403 @@
+
+Data in Holochain is kept in Content Addressable Stores which are key-value stores where the key is the hash of the content. This makes all content self-validating, whether served locally or remotely over the DHT. Data can be retrieved by the record hash, or by the entry hash.
+
+
+The code that comprises a Holochain application is categorized into two different types of zomes: 
+
+1. **Integrity Zomes:** which provide the immutable portion of the app's code that:
+  - identifies the types of entries and links able to be committed in the app, 
+  - defines the structure of data entries, and 
+  - defines the validation code each node runs for each type of entry or link. 
+2. **Coordination Zomes:** which can be modified or added to while an app is live, contain various CRUD operations for entries and links, functions related to following graph links and querying collections of data on the DHT, and any auxillary functionality someone wants to bundle in their application.
+
+Each application running on Holochain is uniquely identified by a DNA hash of the integrity zome code compiled to Web Assembly (WASM) and bundled with additional settings and properties required for that app. 
+
+*Application Note: Multiple DNA-level apps can be bundled together like interoperating micro-services in a larger Holochain Application (hApp), but the locus of data integrity and enforcement remains at the single DNA level, so we will stay focused on that within this document.*
+
+There are three main types of Zome functions:
+1. ( $z _f$ ) zome functions which do not alter state.
+2. ( $Z _f$ ) that can be called to produce state changes, as well as the 
+3. Validation Rules ( $V _R$ ) for enforcing data integrity of any such state changes (additions, modifications, or deletions of data). 
+
+$$
+\begin{aligned}
+ z _{f_1} \dots z _{f_x} &\in \text{Coordinator DNA} \\
+ Z _{f_1} \dots Z _{f_x} &\in \text{Coordinator DNA} \\
+ V _{R_1} \dots V _{R_x} &\in \text{Integrity DNA} 
+\end{aligned}
+$$
+
+*Note about Functions: Most zome calls don't need to be in the immutable mutually enforced rules included in the DNA hash (Integrity Zomes) because since there's no way to verify what functions were involved in producing data, other nodes can only look at the outputs produced to ensure data is structured and validated correctly.*
+
+The first entry of each agent's source chain contains the app DNA. The hash of this initial DNA entry is what demonstrates that each agent possessed identical and complete copies of the the rules by which they will manage and mutually enforce all state changes. If a node has a different DNA hash, then it is in a different network playing by a different set of rules.
+
+**Genesis:** The genesis process for each agent creates three initial entries. 
+
+1. The DNA is stored in the first chain entry with action $C _0$ like this:
```

Comment:
Here you're defining $C_0$ as an *action*, but later you seem to define $C_n$ as a *transaction* comprising many actions